packages/shared/lib/services/connections/credentials/refresh.integration.test.ts

```diff
@@ -0,0 +1,292 @@
+import { NangoError } from '../../../utils/error.js';
+import connectionService from '../../connection.service.js';
+
+describe('refreshOrTestCredentials', () => {
+    beforeAll(async () => {
+        await multipleMigrations();
+        await migrateLogsMapping();
+    });
+
+    it('should not refresh if unauthenticated', async () => {
+        const { env, account } = await seedAccountEnvAndUser();
+        const integration = await createConfigSeed(env, 'unauthenticated', 'unauthenticated');
+        const connection = await createConnectionSeed({ env, provider: 'unauthenticated' });
+        const decryptedConnection = encryptionManager.decryptConnection(connection);
+        if (!decryptedConnection) {
+            throw new Error('Failed to decrypt connection');
+        }
+
+        await wait(2);
+        const onFailed = vi.fn();
+        const onSuccess = vi.fn();
+        const onTest = vi.fn();
+        const res = await refreshOrTestCredentials({
+            account,
+            environment: env,
+            integration,
+            instantRefresh: false,
+            connection: decryptedConnection,
+            onRefreshFailed: onFailed,
+            onRefreshSuccess: onSuccess,
+            connectionTestHook: onTest,
+            logContextGetter: logContextGetter
+        });
+
+        const value = res.unwrap();
+
+        // Nothing should have changed except last_fetched
+        expect(value).toEqual({
+            ...decryptedConnection,
+            last_fetched_at: expect.any(Date)
+        });
+        expect(value.last_fetched_at?.getTime()).toBeGreaterThan(decryptedConnection.last_fetched_at!.getTime());
+        expect(onFailed).not.toHaveBeenCalled();
+        expect(onSuccess).not.toHaveBeenCalled();
+        expect(onTest).not.toHaveBeenCalled();
+    });
+
+    it('should test if api key and succeed', async () => {
+        const { env, account } = await seedAccountEnvAndUser();
+        const integration = await createConfigSeed(env, 'algolia', 'algolia');
+        const connection = await createConnectionSeed({ env, provider: 'algolia', rawCredentials: { type: 'API_KEY', apiKey: 'foobar' } });
+        const decryptedConnection = encryptionManager.decryptConnection(connection);
+        if (!decryptedConnection) {
+            throw new Error('Failed to decrypt connection');
+        }
+
+        await wait(2);
+        const onFailed = vi.fn();
+        const onSuccess = vi.fn();
+        const onTest = vi.fn(() => Promise.resolve(Ok({ tested: true }))) as any;
+        const res = await refreshOrTestCredentials({
+            account,
+            environment: env,
+            integration,
+            instantRefresh: false,
+            connection: decryptedConnection,
+            onRefreshFailed: onFailed,
+            onRefreshSuccess: onSuccess,
+            connectionTestHook: onTest,
+            logContextGetter: logContextGetter
+        });
+
+        const value = res.unwrap();
```

Comment:
nit: I'll call it `refreshed` or something like this. It is a bit more meaningful than 'value'